### ISSUE TITLE
Add `azureMonitor` to `cloud_azure_integrations` resource

### ISSUE THREAD

**Issue by nedl86** [newrelic/terraform-provider-newrelic#2254]
### Feature Description
There's a new Azure Monitor integration available in New Relic ([docs link](https://docs.newrelic.com/docs/infrastructure/microsoft-azure-integrations/azure-integrations-list/azure-monitor/)) which allows for a generic connection between New Relic and Azure Monitor rather than needing to configure an integration per service.

This integration is configurable via Nerd Graph (via the `cloudConfigureIntegration` mutation) and it would be great to see this added to the Terraform provider accordingly.

### Describe Alternatives
No alternatives that I'm aware of, apart from a direct call to Nerd Graph or resorting to click-ops.

### Additional context
One of the benefits of the new integration is simplified configuration, so adding IaC support via Terraform would be the icing on the cake.

**Comment by Cannoli-spec** [comment]
@kidk I'm a Solutions Consultant at New Relic. I see this enhancement has been added to your Q2 roadmap. What is your level of confidence this will be rolled out during that time? Do we have an ETA?

**Comment by pranav-new-relic** [comment]
Hi, this feature has been released with the latest version of Terraform, [v3.22.0](https://github.com/newrelic/terraform-provider-newrelic/releases/tag/v3.22.0). The azureMonitor integration can be configured, using a Terraform Configuration similar to the below.
```
resource "newrelic_cloud_azure_integrations" "foo" {
  linked_account_id = XXXXXX
  account_id        = XXXXXX
  monitor {
    metrics_polling_interval = 1200
    resource_groups          = ["resource_groups"]
    include_tags             = ["env:production"]
    exclude_tags             = ["env:staging", "env:testing"]
    resource_types           = ["microsoft.datashare/accounts"]
  }
}
```
For more details, please head over to the [documentation](https://registry.terraform.io/providers/newrelic/newrelic/latest/docs/resources/cloud_azure_integrations#monitor).
